src/Control/Lens/Internal/Fold.hs

```diff
@@ -72,8 +72,8 @@ instance (Contravariant f, Applicative f) => Monoid (Folding f a) where
 -- The argument 'a' of the result should not be used!
 newtype Traversed a f = Traversed { getTraversed :: f a }
 
-instance Apply f => Semigroup (Traversed a f) where
-  Traversed ma <> Traversed mb = Traversed (ma .> mb)
+instance Applicative f => Semigroup (Traversed a f) where
```

Comment:
I can't help but feel like changing the `Apply` constraint to `Applicative` here (as well as in the `Semigroup` instance for `Sequenced`) is the wrong choice. After all, `Apply` is intended to be `Applicative` without `pure`, so there are more `Apply` instances than `Applicative` ones, meaning that this change would rule out some programs which would otherwise have been permitted.

My inclination would be to adopt the approach that you took for `Effect` here—leave the `Semigroup` instance alone, and add `Apply` constraints to the `Monoid` instances. This is still a breaking change, but one that's much easier to adjust to, since all `Applicative` instances can trivially be made into `Apply` instances.

Reply:
The problem is that `Apply` instances are much less common than `Applicative`s, meaning that the `Semigroup Traversed` instance will become significantly less useful than it currently is. As mentioned by @ekmett in #764, there is a nasty trade-off here.

Reply:
I'm aware of this. But the tradeoff in the other direction is arguable worse, in my opinion. You're demanding that the `f` in this instance not just be an `Apply`, but an `Applicative`. This completely rules out making `f=IntMap`, for example, since `IntMap` is an `Apply` but can't be an `Applicative`.

It's true that there are some `Applicative` instances that aren't `Apply` instances at the moment, but this is easy to retrofit, as `Apply` is just `Applicative` without `pure`.

Reply:
I'm in favor of having the more correct instance using `Apply`. I think that there will be very few actually occurrences where the first argument of one of the `traverseOf_` family of definitions will be constructed using the `Monoid` instance. The work-around would either be to add an `Apply` instance or to write a stand-alone replacement for `mappend` that simply doesn't require the `Apply` instance. It's going to be extremely rare to find some code that both uses `Monoid` to build the `traverseOf_` argument and also requires that that code be built using `Monoid` in particular instead of some local replacement that omits the `Apply` constraint.

Reply:
Fair enough; I can see the argument here. I'm happy to rework this the other way around.